### ISSUE TITLE
Editor complains about "invalid range"

### ISSUE THREAD

**Issue by clecap** [eclipse/wildwebdeveloper#553]
See also:

https://bugs.eclipse.org/bugs/show_bug.cgi?id=567815

When editing Javascript source the editor complains about "invalid range".

Example: "Invalid range: [38718..38707]

The problem shows up consistently across restarts, is difficult to reproduce deterministically but does show up every 5-10 minutes. Looks like it shows up especially when deleting source code.

Here there is a stack trace:

!ENTRY org.eclipse.jface 4 2 2020-10-12 17:30:48.694
!MESSAGE Problems occurred when invoking code from plug-in: "org.eclipse.jface".
!STACK 0
java.lang.IllegalArgumentException: Invalid range: [38718..38707]
	at com.google.common.collect.Range.<init>(Range.java:357)
	at com.google.common.collect.Range.create(Range.java:155)
	at com.google.common.collect.Range.closed(Range.java:180)
	at org.eclipse.lsp4e.outline.SymbolsLabelProvider.getMaxSeverity(SymbolsLabelProvider.java:177)
	at org.eclipse.lsp4e.outline.SymbolsLabelProvider.getImage(SymbolsLabelProvider.java:155)
	at org.eclipse.ui.internal.navigator.NavigatorContentServiceLabelProvider.findImage(NavigatorContentServiceLabelProvider.java:204)
	at org.eclipse.ui.internal.navigator.NavigatorContentServiceLabelProvider.getColumnImage(NavigatorContentServiceLabelProvider.java:112)
	at org.eclipse.ui.internal.navigator.NavigatorContentServiceLabelProvider.getImage(NavigatorContentServiceLabelProvider.java:104)
	at org.eclipse.ui.internal.navigator.NavigatorDecoratingLabelProvider$StyledLabelProviderAdapter.getImage(NavigatorDecoratingLabelProvider.java:63)
	at org.eclipse.jface.viewers.DelegatingStyledCellLabelProvider.getImage(DelegatingStyledCellLabelProvider.java:198)
	at org.eclipse.jface.viewers.DecoratingStyledCellLabelProvider.getImage(DecoratingStyledCellLabelProvider.java:171)
	at org.eclipse.jface.viewers.DelegatingStyledCellLabelProvider.update(DelegatingStyledCellLabelProvider.java:124)
	at org.eclipse.jface.viewers.DecoratingStyledCellLabelProvider.update(DecoratingStyledCellLabelProvider.java:134)
	at org.eclipse.jface.viewers.ViewerColumn.refresh(ViewerColumn.java:144)
	at org.eclipse.jface.viewers.AbstractTreeViewer.doUpdateItem(AbstractTreeViewer.java:959)
	at org.eclipse.jface.viewers.AbstractTreeViewer$UpdateItemSafeRunnable.run(AbstractTreeViewer.java:126)
	at org.eclipse.core.runtime.SafeRunner.run(SafeRunner.java:45)
	at org.eclipse.jface.util.SafeRunnable.run(SafeRunnable.java:174)
	at org.eclipse.jface.viewers.AbstractTreeViewer.doUpdateItem(AbstractTreeViewer.java:1037)
	at org.eclipse.jface.viewers.StructuredViewer$UpdateItemSafeRunnable.run(StructuredViewer.java:427)
	at org.eclipse.core.runtime.SafeRunner.run(SafeRunner.java:45)
	at org.eclipse.jface.util.SafeRunnable.run(SafeRunnable.java:174)
	at org.eclipse.jface.viewers.StructuredViewer.updateItem(StructuredViewer.java:2111)
	at org.eclipse.jface.viewers.AbstractTreeViewer.updateChildren(AbstractTreeViewer.java:2780)
	at org.eclipse.jface.viewers.AbstractTreeViewer.internalRefreshStruct(AbstractTreeViewer.java:1960)
	at org.eclipse.jface.viewers.TreeViewer.internalRefreshStruct(TreeViewer.java:677)
	at org.eclipse.jface.viewers.AbstractTreeViewer.internalRefresh(AbstractTreeViewer.java:1936)
	at org.eclipse.jface.viewers.AbstractTreeViewer.internalRefresh(AbstractTreeViewer.java:1893)
	at org.eclipse.ui.navigator.CommonViewer.internalRefresh(CommonViewer.java:532)
	at org.eclipse.jface.viewers.StructuredViewer.lambda$3(StructuredViewer.java:1484)
	at org.eclipse.jface.viewers.StructuredViewer.preservingSelection(StructuredViewer.java:1400)
	at org.eclipse.jface.viewers.TreeViewer.preservingSelection(TreeViewer.java:363)
	at org.eclipse.jface.viewers.StructuredViewer.preservingSelection(StructuredViewer.java:1361)
	at org.eclipse.jface.viewers.StructuredViewer.refresh(StructuredViewer.java:1484)
	at org.eclipse.jface.viewers.ColumnViewer.refresh(ColumnViewer.java:538)
	at org.eclipse.ui.navigator.CommonViewer.refresh(CommonViewer.java:352)
	at org.eclipse.jface.viewers.StructuredViewer.refresh(StructuredViewer.java:1445)
	at org.eclipse.lsp4e.outline.CNFOutlinePage.labelProviderChanged(CNFOutlinePage.java:300)
	at org.eclipse.jface.viewers.BaseLabelProvider$1.run(BaseLabelProvider.java:75)
	at org.eclipse.core.runtime.SafeRunner.run(SafeRunner.java:45)
	at org.eclipse.jface.util.SafeRunnable.run(SafeRunnable.java:174)
	at org.eclipse.jface.viewers.BaseLabelProvider.fireLabelProviderChanged(BaseLabelProvider.java:72)
	at org.eclipse.ui.internal.decorators.DecoratorManager$1.run(DecoratorManager.java:349)
	at org.eclipse.core.runtime.SafeRunner.run(SafeRunner.java:45)
	at org.eclipse.ui.internal.decorators.DecoratorManager.fireListener(DecoratorManager.java:346)
	at org.eclipse.ui.internal.decorators.DecorationScheduler$3.runInUIThread(DecorationScheduler.java:525)
	at org.eclipse.ui.progress.UIJob.lambda$0(UIJob.java:95)
	at org.eclipse.swt.widgets.RunnableLock.run(RunnableLock.java:40)
	at org.eclipse.swt.widgets.Synchronizer.runAsyncMessages(Synchronizer.java:185)
	at org.eclipse.swt.widgets.Display.runAsyncMessages(Display.java:4124)
	at org.eclipse.swt.widgets.Display.readAndDispatch(Display.java:3791)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine$5.run(PartRenderingEngine.java:1157)
	at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:338)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.run(PartRenderingEngine.java:1046)
	at org.eclipse.e4.ui.internal.workbench.E4Workbench.createAndRunUI(E4Workbench.java:155)
	at org.eclipse.ui.internal.Workbench.lambda$3(Workbench.java:644)
	at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:338)
	at org.eclipse.ui.internal.Workbench.createAndRunWorkbench(Workbench.java:551)
	at org.eclipse.ui.PlatformUI.createAndRunWorkbench(PlatformUI.java:153)
	at org.eclipse.ui.internal.ide.application.IDEApplication.start(IDEApplication.java:150)
	at org.eclipse.equinox.internal.app.EclipseAppHandle.run(EclipseAppHandle.java:203)
	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.runApplication(EclipseAppLauncher.java:134)
	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.start(EclipseAppLauncher.java:104)
	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:401)
	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:255)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:567)
	at org.eclipse.equinox.launcher.Main.invokeFramework(Main.java:657)
	at org.eclipse.equinox.launcher.Main.basicRun(Main.java:594)
	at org.eclipse.equinox.launcher.Main.run(Main.java:1465)

**Comment by mickaelistria** [comment]
Thanks.
Can you please enable Language Server logs for JavaScript via Preferences; and when this issue occurs, share the last language server related to "symbols" from the log file?

**Comment by clecap** [comment]
Thanks for the response and sorry for the delay. Due to this but I have not used eclipse lately and rather used Atom for js work on long files. On smaller files it does not show up in eclipse. However, I now caught another case. I shall attach a trimmed down version of the log, since the original log was quite long. Also I attach a version of the file at the moment the error occured.

[org.eclipse.wildwebdeveloper.jsts copy.log](https://github.com/eclipse/wildwebdeveloper/files/5625588/org.eclipse.wildwebdeveloper.jsts.copy.log)

File was:



const fs = require('fs').promises;


var Module = require('module');

var babelJsx = require("@babel/core").transform("code", {
  plugins: ["@babel/plugin-transform-react-jsx"],
});

function install () {

Module.prototype.require = new Proxy( Module.prototype.require, {
  apply: async function (target, thisArg, argumentsList) {
    let name = argumentsList[0];
    console.log ("called my require on " + name);
    if( name.endsWith(".jsx") ) {
      console.log ("requiring a jsx at: " + name);    
      const code = await fs.readFile(name, "utf8");
      console.log (code);

      return new Promise ( (resolve, reject) => {
           babelJsx.transform(code, {} , function(err, result) {
  result; // => { code, map, ast }
    });
    
       
      } );
  
    }
    return Reflect.apply(target, thisArg, argumentsList)
  }
})

}

exports.install = install;

**Comment by mickaelistria** [comment]
Wild Web developer got a new release yesterday; including update of language servers which may fix the issue. Can you please try latest release and report whether it works better ?

**Comment by clecap** [comment]
Tried it and to ensure proper infrastructure, downloaded a completely fresh installation of eclipse. The good news: The problem seems to be gone.

The bad news: 95% of all the possibilities of configuring the editor also are gone. In particular the customization possibilities of autocompletion. :-( 

Or did I miss something on my upgrade?

**Comment by mickaelistria** [comment]
Closing as initial issue is fixed.
Those customization were never provided by Wild Web Developer; they were provided by legacy WebTools JSDT or XML editors. Wild Web Developer never honored them.

**Comment by clecap** [comment]
??? Honestly? This is 2020? Yes? Hello??

Well. I asked you answered. Thank you.

But...very much disappointed by the tremendous reduction in features. If you care for feedback: Why would one want to use an editor which is so much reduced in features...if there is a good reason, currently the web pages fails to point this out. Practically every competing system offers them.

**Comment by o-alquimista** [comment]
I can reproduce this with Eclipse PDT 2021-03 (4.19).

![Captura de tela de 2021-03-23 14-16-44](https://user-images.githubusercontent.com/25424263/112189515-d9ed2800-8be2-11eb-8bd2-3bdea99a6335.png)

I have time to help diagnose this. I use Eclipse a lot. What can I do to help?

**Comment by mickaelistria** [comment]
@o-alquimista Can you please share the file you were using when this happens? provide details about how to reproduce? Share content of your <workspace>/.metadata/.log file? Share the version of Wild Web Developer you have installed?...

**Comment by o-alquimista** [comment]
Below is the gist containing one of the files I was editing when it happened, and the log file you requested. The log contains about 18 instances of that "Invalid range" error.
https://gist.github.com/o-alquimista/5db0b43e700c7b211ec3d435ace9fb17

## Eclipse installation
Wild Web Developer 0.10.6.202102181620

I used the Eclipse installer to install PDT in my home directory. The Java runtime was downloaded by the installer itself. I just picked the latest version.

## Reproducing it
It usually happens when I'm creating or editing an `if/else` block. I'm not sure that's the only way to trigger it, though.

```js
if (condition === true) {
// this is the point where the error often pops up, before the if block is closed
```

It's so annoying and persistent I don't even close the little error window - I just minimize it; it keeps popping up and stealing focus from the editor.

**Comment by mickaelistria** [comment]
I've opened https://git.eclipse.org/r/c/lsp4e/lsp4e/+/178365 to LSP4E to avoid such issues.

**Comment by o-alquimista** [comment]
> I've opened https://git.eclipse.org/r/c/lsp4e/lsp4e/+/178365 to LSP4E to avoid such issues.

Do you have any idea when will these fixes be available to us? I'm on Eclipse 2021-06 (4.20.0) and the bug is still there.

Tip for other users: this issue is even worse on the Xfce desktop, because the little error dialog keeps coming back into focus all the time!

**Comment by mickaelistria** [comment]
> Do you have any idea when will these fixes be available to us? 

There are still important pending item before LSP4E produces a new release, but you can install the snapshot version from https://download.eclipse.org/lsp4e/snapshots to already get this fix before it's released.
